xml_converter/generators/generate_cpp.py

```diff
@@ -260,71 +269,73 @@ def generate_cpp_variable_data(
         cpp_includes.cpp_absolute_includes.add("type_traits")
 
     for filepath, document in sorted(data.items()):
-        fieldval = document.metadata
-        attribute_name: str = attribute_name_from_markdown_data(fieldval['name'])
+        fieldval = document.metadata_dataclass
+        attribute_name: str = attribute_name_from_markdown_data(fieldval.name)
 
-        if doc_type in fieldval['applies_to']:
+        if doc_type in fieldval.applies_to_as_str():
             xml_fields: List[str] = []
             side_effects: List[str] = []
             write_to_xml: bool = True
             default_xml_field: str = ""
 
-            if fieldval['type'] in documentation_type_data:
-                cpp_type = documentation_type_data[fieldval['type']]["cpp_type"]
-                class_name = documentation_type_data[fieldval['type']]["class_name"]
+            if fieldval.variable_type in documentation_type_data:
+                cpp_type = documentation_type_data[fieldval.variable_type]["cpp_type"]
+                class_name = documentation_type_data[fieldval.variable_type]["class_name"]
                 cpp_includes.cpp_relative_includes.add("attribute/{}.hpp".format(class_name))
 
-            elif fieldval['type'] in ["Custom", "CompoundCustomClass"]:
-                cpp_type = fieldval['class']
-                class_name = insert_delimiter(fieldval['class'], delimiter="_")
+            # elif fieldval.variable_type in ("Custom", "CompoundCustomClass"):
```

Comment:
This comment seems to be an alternative to 287. Should it be deleted?

Reply:
I left this here because the code in 287 works for type inference, but for some reason the code in 286 does not. They should be equal and I like the style of 286 better.

This might not be an issue in mypy but it is in the LSP I am using for my editor pyright. I think both the comment and the "working" line should stay.